app/resources/cdl/charge_manager.rb

```diff
@@ -84,8 +58,47 @@ def create_hold(netid:)
       change_set_persister.save(change_set: change_set)
     end
 
-    def available_item_id
-      (item_ids - resource_charge_list.charged_items.map(&:item_id)).first
+    def activate_holds!
+      @resource_charge_list = expire_holds!
+      return unless available_charge_slot? && resource_charge_list.pending_or_active_holds.present?
+      return unless activate_hold_count.positive?
+      activated_holds = Array.new(activate_hold_count).map do
+        hold = resource_charge_list.pending_holds.first
+        hold.expiration_time = 1.hour.from_now
```

Comment:
where is this saved?

Reply:
The holds are mutated in place and then saved below.

Reply:
okay they're held by reference in the hold_queue array.